src/Chainweb/CutDB/Sync.hs

```diff
@@ -103,11 +103,18 @@ syncSession v useOrigin p db logg env pinf = do
         logg @T.Text Debug $ "put cut " <> encodeToText c
 
     receive = do
+        cur <- _cut db
+
         -- Query cut that is at most 1000 blocks ahead
-        h <- _cutHeight <$> _cut db
-        c <- getCut cenv (h + min catchupStepSize (int farAheadThreshold - 1))
-            -- Cf. documentation of 'fastAheadThreshold' for why this bound is
-            -- needed
+        let curHeight = _cutHeight cur
+            curChainCount = length $ view cutMap cur
```

Comment:
Is this a question we ask a lot? Wondering if this should be a util function ... fine to do in a later PR

Reply:
hmm, not sure how often it comes up in the code. The use here won't count, since it should be replaced by a block height based limit. C.f. the comment on the next line.